src/index.d.ts

```diff
@@ -0,0 +1,608 @@
+	SpotLight,
+	WebGLArrayRenderTarget,
+	Camera,
+	PerspectiveCamera,
+	Loader,
+	MeshStandardMaterial,
+	Light,
+	Material,
+	Object3D,
+	Texture,
+	Vector2,
+	WebGLRenderer,
+	WebGLRenderTarget,
+	BufferGeometry,
+	Matrix4,
+	Color,
+	ShaderMaterialParameters,
+	Vector4,
+	MeshStandardMaterialParameters,
+	Spherical,
+	Scene,
+	RenderTargetOptions,
+	PMREMGenerator,
+	DataArrayTexture,
+	BufferAttribute,
+} from 'three';
+import {
+	MeshBVH,
+	MeshBVHOptions,
+	MeshBVHUniformStruct,
+	UIntVertexAttributeTexture,
+} from 'three-mesh-bvh';
+
+//#region core
+
+export class PathTracingRenderer<TMaterial extends ShaderMaterial> {
+
+	constructor( renderer: WebGLRenderer );
+
+	readonly samples: number;
+	readonly target: WebGLRenderTarget;
+
+	material: TMaterial | null;
+	alpha: boolean;
+	camera: Camera | null;
+	tiles: Vector2;
+	stableNoise: boolean;
+	stableTiles: boolean;
+
+	setCamera( camera: Camera ): void;
+	setSize( width: number, height: number ): void;
+	getSize( target: Vector2 ): void;
+
+	dispose(): void;
+	reset(): void;
+	update(): void;
+
+}
+
+export class QuiltPathTracingRenderer<TMaterial extends ShaderMaterial> extends PathTracingRenderer<TMaterial> {
+
+	quiltDimensions: Vector2;
+	viewCount: number;
+	viewCone: number;
+	viewFoV: number;
+	displayDistance: number;
+	displayAspect: number;
+
+	setFromDisplayView( viewerDistance: number, displayWidth: number, displayHeight: number ): void;
+
+}
+
+export const NO_CHANGE: 0;
+export const GEOMETRY_ADJUSTED: 1;
+export const GEOMETRY_REBUILT: 2;
+export type ChangeType = typeof NO_CHANGE | typeof GEOMETRY_ADJUSTED | typeof GEOMETRY_REBUILT;
+
+export interface StaticGeometryGeneratorResult {
+
+	changeType: ChangeType
+	materials: Array<Material>;
+	geometry: BufferGeometry;
+
+}
+
+declare class StaticGeometryGenerator {
+
+	constructor( objects?: Object3D | Array<Object3D> );
+
+	objects: Array<Object3D> | null;
+	useGroups: boolean;
+	applyWorldTransforms: boolean;
+	generateMissingAttributes: boolean;
+	overwriteIndex:boolean;
+	attributes: Array<string>;
+
+	setObjects( objects: Object3D | Array<Object3D> ): void;
+
+	generate( targetGeometry?: BufferGeometry ): StaticGeometryGeneratorResult;
+
+}
+
+export interface PathTracingSceneGeneratorResult {
+
+	bvhChanged: boolean;
+	bvh: MeshBVH;
+	lights: Array<Light>;
+	iesTextures: Array<DataTexture>;
+	geometry: BufferGeometry;
+	materials: Array<Material>;
+	textures: Array<Texture>;
+	objects: Array<Object3D>;
+
+}
+
+export class PathTracingSceneGenerator {
+
+	constructor( objects?: Object3D | Array<Object3D> );
+
+  readonly initialized: boolean;
+
+  bvhOptions: MeshBVHOptions;
+	attributes: Array<string>;
+	generateBVH: boolean;
+
+	bvh: MeshBVH | null;
+	geometry: BufferGeometry;
+	staticGeometryGenerator: StaticGeometryGenerator;
+
+	setObjects( objects: Object3D | Array<Object3D> ): void;
+	// TODO: The worker and its type are not exported from `three-mesh-bvh`
+	setBVHWorker( bvhWorker: any ): void;
```

Comment:
`three-mesh-bvh` doesn't export the BVH workers or types for them. Should this type just be declared locally?

Reply:
Yeah unfortunately using WebWorkers can break a lot of bundle systems. We can define an interface here, though, that the object needs to adhere to. Something like this:

```ts
{
  generate( geometry : BufferGeometry, bvhOptions: MeshBVHOptions ) : Promise<MeshBVH>
}
```